tests/inference/test_serve.py

```diff
@@ -0,0 +1,139 @@
+            hpus_per_worker,
+            deepspeed,
+            workers_per_group,
+            ipex,
+            device,
+            serve_local_only,
+            simple,
+            keep_serve_termimal,
+        )
+        for config_file in config_file_array
+        for model_id_or_path in model_id_or_path_array
+        for models in models_array
+        for port in port_array
+        for route_prefix in route_prefix_array
+        for cpus_per_worker in cpus_per_worker_array
+        for gpus_per_worker in gpus_per_worker_array
+        for hpus_per_worker in hpus_per_worker_array
+        for deepspeed in deepspeed_array
+        for workers_per_group in workers_per_group_array
+        for ipex in ipex_array
+        for device in device_array
+        for serve_local_only in serve_local_only_array
+        for simple in simple_array
+        for keep_serve_termimal in keep_serve_termimal_array
+    ],
+)
+def test_script(
+    config_file,
+    model_id_or_path,
+    models,
+    port,
+    route_prefix,
+    cpus_per_worker,
+    gpus_per_worker,
+    hpus_per_worker,
+    deepspeed,
+    workers_per_group,
+    ipex,
+    device,
+    serve_local_only,
+    simple,
+    keep_serve_termimal,
+):
+    cmd_serve = ["python", "../inference/serve.py"]
+    if config_file is not None:
+        cmd_serve.append("--config_file")
+        cmd_serve.append(str(config_file))
+    if model_id_or_path is not None:
+        cmd_serve.append("--model_id_or_path")
+        cmd_serve.append(str(model_id_or_path))
+    if models is not None:
+        cmd_serve.append("--models")
+        cmd_serve.append(str(models))
+    if port is not None:
+        cmd_serve.append("--port")
+        cmd_serve.append(str(port))
+    if route_prefix is not None:
+        cmd_serve.append("--route_prefix")
+        cmd_serve.append(str(route_prefix))
+    if cpus_per_worker is not None:
+        cmd_serve.append("--cpus_per_worker")
+        cmd_serve.append(str(cpus_per_worker))
+    if gpus_per_worker is not None:
+        cmd_serve.append("--gpus_per_worker")
+        cmd_serve.append(str(gpus_per_worker))
+    if hpus_per_worker is not None:
+        cmd_serve.append("--hpus_per_worker")
+        cmd_serve.append(str(hpus_per_worker))
+    if deepspeed:
+        cmd_serve.append("--deepspeed")
+    if workers_per_group is not None:
+        cmd_serve.append("--workers_per_group")
+        cmd_serve.append(str(workers_per_group))
+    if ipex:
+        cmd_serve.append("--ipex")
+    if device is not None:
+        cmd_serve.append("--device")
+        cmd_serve.append(str(device))
+    if serve_local_only:
+        cmd_serve.append("--serve_local_only")
+    if simple:
+        cmd_serve.append("--simple")
+    if keep_serve_termimal:
+        cmd_serve.append("--keep_serve_termimal")
+
+    result_serve = subprocess.run(cmd_serve, capture_output=True, text=True)
+
+    # TODO: Find a better way to assert the result, like checking processes etc.
```

Comment:
It's better to print out the result if we want to check it in the CI before we find a better way to assert.

Reply:
added

Reply:
In addition to check "Error" in the stderr, I think you may also need check the exit code of the launched processes. Successful run is 0. If exception thrown, it should be non-zero. Just may sure we can capture all the errors in the CI. could you try that?

Reply:
Sure, returncode assert added and CI passed.